python/python/tests/test_async_client.py

```diff
@@ -5562,6 +5562,89 @@ async def test_xack(
         with pytest.raises(RequestError):
             await redis_client.xack(string_key, group_name, [stream_id1_0])
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_xgroup_set_id(
+        self, redis_client: TRedisClient, cluster_mode, protocol, request
+    ):
+        key = f"{{testKey}}:{get_random_string(10)}"
+        non_existing_key = f"{{testKey}}:{get_random_string(10)}"
+        string_key = f"{{testKey}}:{get_random_string(10)}"
+        group_name = get_random_string(10)
+        consumer_name = get_random_string(10)
+        stream_id0 = "0"
+        stream_id1_0 = "1-0"
+        stream_id1_1 = "1-1"
+        stream_id1_2 = "1-2"
+
+        # setup: create stream with 3 entries, create consumer group, read entries to add them to the Pending Entries
+        # List
+        assert (
+            await redis_client.xadd(key, [("f0", "v0")], StreamAddOptions(stream_id1_0))
+            == stream_id1_0
+        )
+        assert (
+            await redis_client.xadd(key, [("f1", "v1")], StreamAddOptions(stream_id1_1))
+            == stream_id1_1
+        )
+        assert (
+            await redis_client.xadd(key, [("f2", "v2")], StreamAddOptions(stream_id1_2))
+            == stream_id1_2
+        )
+        assert await redis_client.xgroup_create(key, group_name, stream_id0) == OK
+        assert await redis_client.xreadgroup({key: ">"}, group_name, consumer_name) == {
+            key: {
+                stream_id1_0: [["f0", "v0"]],
+                stream_id1_1: [["f1", "v1"]],
+                stream_id1_2: [["f2", "v2"]],
+            }
+        }
+        # sanity check: xreadgroup should not return more entries since they're all already in the Pending Entries List
+        assert (
+            await redis_client.xreadgroup({key: ">"}, group_name, consumer_name) is None
+        )
+
+        # reset the last delivered ID for the consumer group to "1-1"
+        # ENTRIESREAD is only supported in Redis version 7.0.0 and above
+        if await check_if_server_version_lt(redis_client, "7.0.0"):
+            assert await redis_client.xgroup_set_id(key, group_name, stream_id1_1) == OK
+        else:
+            assert (
+                await redis_client.xgroup_set_id(
+                    key, group_name, stream_id1_1, entries_read_id=stream_id0
+                )
+                == OK
+            )
+
+            # XGROUP SETID accepts "0" for the entries read ID, but does not accept "0-0"
+            with pytest.raises(RequestError):
+                await redis_client.xgroup_set_id(
+                    key, group_name, stream_id1_1, entries_read_id="0-0"
```

Comment:
maybe change this:
```suggestion
                    key, group_name, stream_id1_1, entries_read_id="stream_id0"
```
since the documentation says it cannot be the first or zero entry.